### ISSUE TITLE
IDEATION: Order of phases for Round 2 testing? Add in a discussion Q for Round 2? 

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1289]
@ilmartyrk @kevincrepin 

One quite big thing to decide before we do the round 2 testing: now that we have the topics (and everything related to phases) set up in the order of 'ideation > discussion > voting', do we keep it this way until we have the 'flexible phases' solution (if that's the way we go with it) or until we see very strong evidence to switch it, or do we change it already and re-test it in the other order in Round 2? 

I guess the main questions for us to consider would be:
- Is it technically a lot of work to switch them, and is it doable by the w/c 17th, or worth delaying for if not? (I imagine a lot of bugs could be thrown up by doing so?) 
- Do we feel we have enough evidence that it's preferable to switch it, for it to be worth the time and bug fixing for now? (Note to self that we should do more prototype testing in future, to avoid this situation :))  
- Are there any simpler solutions that would make the current order work better, as a temporary 'quick fix'?  

Some thoughts on these questions from me: 

If it was easy to switch it, I would probably err towards switching it back to discussion > ideation > voting from what I've seen of the use cases so far, for these reasons:

- The 3 use cases I've seen try to use the phases in the order they are have been interested in using the discussion phase where it is, but the way it's currently set up (with no discussion question / prompt, and with only 1 space, not multiple spaces in a more forum-like setup) doesn't really work for what they're expecting to do there. It also seems their current thinking is mostly (though not always) to discuss the ideas in more detail, but their is space under each idea for discussion. 

- In all 3 of the use cases, it has seemed that having a discussion before ideation about whether or not to even go ahead with the ideation would have been helpful. As in all 3 use cases, at some point during the ideation, the question of whether the ideation was even relevant has been raised.  For example:

(1) Before ideating on what do with the community space: "Should we do something new with X space?", if yes, then "What should we do with it?". 
(2) Before ideating on membership criteria: "Should we have strict criteria for membership?", if yes, then "What should the criteria be?". 
(3) Before ideating on how to overhaul the Youth Employment Camp for 2024: "Should we run the Youth Employment Camp in 2024 or have a year to redevelop it?".

However, it seems to me that if we were setting up the phases for this flow, then it would be better as "Discussion, vote, ideation, vote." So that rather than just a pro and con discussion about whether to do X, participants could also vote, before the ideation then happens if the vote was in favour.  

This erring towards switching it is really just guesswork though, and I have some doubts. E.g. if the need for a pro-con discussion on whether to go ahead comes up during the discussion, they wouldn't have the possibility to add a discussion phase, if they had started with ideation with the phases switched so it is second. They may also happily split this into two topics: discuss + vote, then ideate + vote.  

If we were going to leave it as it is for now, then I think we could add in the discussion prompt/s we've talked about [here](https://github.com/citizenos/citizenos-fe/issues/251) (and in the call), to make it at least function a little better, even if not quite the ideal.  

Re the insights on flexible phases and multiple discussions/votes/ideations per topic/phase, I have a few more insights to add to the [Google doc]((https://docs.google.com/document/d/1oiaPxNqRfyaVmU7FhficcRA9TiUW7ISoHzti8Eh_61I/edit?usp=sharing) that came up in participant tests. I'll add them in a few days as I assume we won't be getting this ready for Round 2 testing, but can say that the insights feel pretty conclusive that flexibility and multiplicity are needed :))

**Comment by kevincrepin** [comment]
How I see this is that either we go with the fixed linear process because we have a very good argument as to why doing it in such an order would be the most beneficial – based on experience running these kind of sessions irl by @loorm , I would think. He had a specific reason as to why Ideation first makes sense, so until we have (if we have) dynamic phases, I would leave it like this.

Or we don't set any rules at all and we go with the dynamic approach which would allow anyone to use it however they want. But there's a reason why people like @loorm run workshops. Probably because doing it "however you want" does not always yield the best results :D 

Imho, this is a core principle that should be decided within the Citizen OS team as to what it stands for. I honestly don't know which would be the best way.

**Comment by BeccaMelhuish** [comment]
OK yes I agree - we're probably not in a position to make this decision yet, and we can come back to it once we've done more research into the needs of different use cases (as the one Margo described was just one particular process - so maybe for that case this order is correct, but for others it doesn't seem to be), and have decided more clearly internally what our priority target group/use is and researched their processes :) 

Though for now, as it will be a while until the decision on whether or not the dynamic approach is the way forward is made (and then designed and tested and developed if so), shall we have an interim solution? 

So for example, when an admin clicks this "Start topic discussion" button:

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/8d6d80eb-dbd6-4451-b1e8-a906b0004855)

And gets this pop-up, then there they would have a second tab which allows them to add a discussion question/prompt (and deadline, potentially?): 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/5a33c94a-bdda-48f5-9c51-8b0386ce0b62)

This would be a short term solution to the problem of each phase (in some cases) needing a different text, and make the discussion phase more usable in the order the phases are currently in :)  

Though if we are adding it to discussions when discussions are the second phase, we should also include it when discussions are the first phase (i.e. in 'discussion topic' create flow). In that situation it's less needed though, so might be repetitive, and perhaps could therefore be optional?

**Comment by ilmartyrk** [comment]
@BeccaMelhuish @kevincrepin @loorm the current discussion is meant to be based on the topic content so adding the question would complicate it. "Planning of a playground", but the discussion question is set "What material we should use for the swings?" that would confuse users a lot as there would be people who would like to discuss some other questions that are related to the topic too.

**Comment by kevincrepin** [comment]
We floated the idea of having several discussion "widgets" (similar to threads in a forum) and a similar approach would be taken for ideation and voting then. So then the prompt or question or "title" would be making sense (and needed).

Then there can be different discussions taking place about one text, but with a specific focus.

**Comment by BeccaMelhuish** [comment]
Yes agreed @kevincrepin, though even if we only have one discussion per topic (which will probably be the case when we launch ideation?), from watching the user tests and running some mock processes myself in order to test the participants role, it seems the discussion prompt is definitely needed. 

Here are some examples: 

1. Youth Centre manager is running a process about what to do with the annual Employment Camp. She starts with ideation, and writes the topic description all about ideation. She then moves the topic into discussion, but the text is the same, from the ideation. So the text is all about adding ideas, but she doesn't want people to be adding ideas. She doesn't notice that this would be confusing for participants, and hasn't given any guidance to them about what the discussion is actually about (which she was expecting to be more general, not on ideas): 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/ca1527cb-fb84-475c-91db-13c01a8bbbfa)

More on this in [this issue](https://github.com/citizenos/citizenos-fe/issues/251). 

2. I was pretending to be an admin for a process to decide what to do with the local village green. I started with ideation, and wrote the text all about ideation. Then when it came to the discussion phase, I deleted the whole text and replaced it with the discussion prompt. (So essentially, I found a "hack" to add a discussion prompt, but not all admins would think to do this, and this also meant that the text about what the whole process was about was then lost). And in this case the question was hidden under 'Read more', and wasn't seen by the user until right at the end of the test (in follow-up phase). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/59dac1ad-7fec-49a8-b042-59595f686233)
![image](https://github.com/citizenos/citizenos-fe/assets/61320526/0bd5183f-b245-4a49-8c49-fa602f02ca71)

3. I was pretending to be an admin for a process to decide the membership criteria for a collective. I started with ideation, and wrote the text all about ideation. Then when it came to the discussion phase, I had to edit the text like this. (Again, I found a "hack" to add a discussion prompt, but not all admins would think to do this. It also meant this text showed for the voting part, which was then confusing, and I couldn't edit it out again as it was locked for voting.)

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/4f183527-11c6-4a18-8d6f-1020bf9fbb72)

**Comment by BeccaMelhuish** [comment]
So my proposal for the timescale split would be: 

- **Ideation launch fixes:** Add the discussion prompt (at least when it is the second phase in a topic, it makes less sense when it's the first phase, as Ilmar says - they are discussing on the topic text). And discussion deadline just for consistency if possible, but less crucial. 

- **Bigger structural fixes:** All things relating to changes to the order or the flexibility or multiple widgets within each phase. _**Unless**_ it is actually very easy (technically and design-wise) to add multiple discussion widgets to a discussion phase already, in which case, let's do it and test it in Round 2 :)

**Comment by BeccaMelhuish** [comment]
@kevincrepin @ilmartyrk 

I'll add a couple of additional insights in favour of adding a discussion prompt to the discussion phase already (potentially in general, not just for when it's the second phase). Though please [see comment above here](https://github.com/citizenos/citizenos-fe/issues/1289#issuecomment-2152637166) for the admin ones :) 

**Insights from Ideation Round 1 participant UX tests**

- One user (in the discussion phases of an ideation topic) was looking missing a 'clear statement' so that she knew which way was pro and which was con. So if we have a clear discussion Q, we can prompt admins to formulate it in a way that makes it clear to participants what pro is and what con is. 

- Users aren't always opening the 'Read more'. That's partly a separate issue (now made [here](https://github.com/citizenos/citizenos-fe/issues/1300)), but if the discussion question is far down in the topic text, they won't (currently, as it is with the design now) always actually see it.

**Comment by kevincrepin** [comment]
About the discussion prompt. If it shows there's a need I don't have anything against it. I'd redesign the other "widgets" also a little bit so they're all consistent:
![Screenshot 2024-06-10 at 15 02 19](https://github.com/citizenos/citizenos-fe/assets/26792281/b3b2716f-ef40-418e-b380-cc93d4c5a2ce)

And then when creating a discussion topic we need to introduce an additional step after Settings like we have in ideation- But then adjusted for discussion:
![Screenshot 2024-06-10 at 15 04 25](https://github.com/citizenos/citizenos-fe/assets/26792281/1b7945b5-372a-41f6-a74d-0c62c464bac4)

**Comment by BeccaMelhuish** [comment]
Great. And how about the 'Start topic discussion' dialogue box (and steps) [here](https://github.com/citizenos/citizenos-fe/issues/1289#issuecomment-2152283187)? 

For the copy, what do you think @kevincrepin do we call it a discussion question or a discussion prompt?

**Comment by kevincrepin** [comment]
I'm not yet used to discussion being the second step :) So when creating a discussion phase after ideation, the popup (at least for now) could be like this:
![Screenshot 2024-06-10 at 15 53 01](https://github.com/citizenos/citizenos-fe/assets/26792281/b8c50f28-79ed-4f96-b28e-5d5088ec00b1)

I think a "prompt" makes more sense than a question because a prompt _can_ also be a question if I'm not mistaken. But not sure if this is a known word for everyone though...

**Comment by BeccaMelhuish** [comment]
Great, this looks good! I'd maybe take out the 'topic' here though, wdyt?

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/1e45d92f-1d38-42f0-90e9-92dfe2dfd927)

I agree yes a prompt could be both so that's good, though good point re the accessibility of the language. @oksks do you have a feeling on that?

**Comment by kevincrepin** [comment]
@ilmartyrk I added mobile views for:

- "widget" updated with prompt: https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=13511-112444&t=5P8GCGSpqoB12ZBy-1
- Discussion phase creation dialog window: https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=13357-106741&t=5P8GCGSpqoB12ZBy-1

**Comment by oksks** [comment]
@BeccaMelhuish - "Start discussion phase" sounds good and translates well:)

**Comment by BeccaMelhuish** [comment]
Grand thank you @oksks! And what do you think about this bit below - is 'prompt' accessible enough language? 

'Discussion question' would be easier to understand, but prompt is more accurate - as a discussion isn't always based on a question.. Our only doubt is how clear it might be to non-native speakers.. 

Though @kevincrepin maybe we could add a tooltip here that makes it clear to anyone unsure of the word 'prompt', and also mentions that they should make sure it's framed in a way that's clear which is pro and which is con?  

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/8229eaf1-7993-4a56-b3be-786b6eb984ee)

**Comment by kevincrepin** [comment]
We can have a tooltip here, sure.

But I have to say I'm a bit starting to worry that if our guideline to this mandatory field already uses words people can't understand, how will they be able to come up with this prompt itself...

**Comment by BeccaMelhuish** [comment]
@ilmartyrk Have tried it out and here's what I got so far :)

- Deadline in the 'discussion' part of the create flow has gone a bit awry:

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/3e07e3af-b86b-4330-9d12-026d06c3ad2d)

- @kevincrepin I'm wondering what should happen when a discussion closes? We've not discussed this bit..

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/b23f4da9-465b-4253-940f-daaea5ad0932)

- Preview should show the prompt (and deadline if set, however that bar will be in the published version):

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/1801deba-43d2-4a65-8a10-ec2d21486d22)

- When I tried to publish, got this error here (also made a [separate issue](https://github.com/citizenos/citizenos-fe/issues/1312)):

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/668ccb57-1703-4ac4-97b0-b6ad7a3c215b)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish posted fixes

**Comment by oksks** [comment]
> Grand thank you @oksks! And what do you think about this bit below - is 'prompt' accessible enough language?
> 'Discussion question' would be easier to understand, but prompt is more accurate - as a discussion isn't always based on a question.. Our only doubt is how clear it might be to non-native speakers..

Let's not use "prompt", as I agree, it's hard to understand and even harder to translate accurately. 
Maybe say **"Please add a specific discussion topic"**?, instead of "prompt"? Would that have more or less the same outcome?

And re: the string 
"Please add a specific discussion prompt to bring focus to participants", could we maybe rephrase it to => 
**"Please add a specific discussion topic to help guide the participants"**?

**Comment by BeccaMelhuish** [comment]
@oksks hmm I'm not sure about the use of 'topic' myself, as that's what we call the whole thing..  

I can't think of anything that works better than prompt though! It does cover all options for what they'd add here really well. 

If it was only the understandability of the word, I think the tooltip would solve it. But if it's also going to make it hard for translation, then maybe we go for 'question or statement'?

So: 

**"Please add a specific discussion question or statement to help guide the participants"**

Or, we just go with 'discussion question' and assume that if they don't want to write a question they'll just not, they'll use their initiative to write a statement or 'prompt' instead.. 

Maybe we're overthinking it :)

**Comment by BeccaMelhuish** [comment]
@kevincrepin @oksks I've changed it to question for now and let's see if it causes any issues in the tests :)

**Comment by BeccaMelhuish** [comment]
@ilmartyrk I was testing this issue, and everything worked fine up until the point I tried to post the first argument, when I got this 'Internal Server Error' again:
(Tried it in incognito too, in the same topic, and got the error at the same point). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/d6a1dd36-9c9a-416e-825b-642847395594)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish was an issue related to database migrations. Should work now

**Comment by BeccaMelhuish** [comment]
Great, seems to be working for me indeed, will close this issue now then!

**Comment by BeccaMelhuish** [comment]
Re-opening as just spotted something:

Might be connected to [this issue](https://github.com/citizenos/citizenos-fe/issues/1192), but when a discussion has been added but there are no arguments yet, the discussion question chunk needs to show up already. Currently it only shows up once you have posted an argument. It should appear just above the 'Get the discussion started' text, when that text is there (which as per the other issue, should be only when the admin has actually added a discussion phase (currently incorrect). 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/d1bf6598-2320-4ae4-90f8-b58a563d77ee)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish posted a fix
